.github/workflows/nightly_run.yml

```diff
@@ -0,0 +1,622 @@
+# Copyright (C) 2019 Intel Corporation.  All rights reserved.
```

Comment:
It looks like most of the code here is in common with other jobs, any easy way to avoid this duplication?

Reply:
will try

Reply:
i've checked about github actions reusage, the only thing is https://github.com/actions/starter-workflows/issues/245
for me the only thins is they allow now, is to have a "template" in other file to call it with different inputs (eg the other file is one/multiples job which is called with different inputs)
we could do this like
```
prepare:
  uses ./.github/common.yml
  ```
but then we lose needs functionality
if later you write
```
stepN:
  needs[build_llvm]
```
you will get the error because this job is not actually in this file and it can't find it's name
as I know "waterfall" (or fallback) syntaxis is not supported, otherwise we could do
```
j1:
j2:
j3:
j4:
  uses ./.github/common.yml
```
In short, don't see any pretty enough way

Please correct me if others have any good ideas about it

Reply:
we used to apply [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) in binary releasing scripts. https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/.github/workflows/release_process.yml#L56

It can help to handle some of duplications but not a silver bullet for all of them.

Reply:
yeah, it applies there because there's just one job triggered with different input. I don't think it will add much sense for us unfortunately (it's nice if you have only one job in file, if you have 2+ every `uses` instruction will trigger them all) so I'd not do that in this PR at least